index.js

```diff
@@ -1,82 +1,135 @@
 'use strict';
 
-function constant(value) {
-    return function() {
-        return value;
+var GLOBAL_ATTRIBUTE_PREFIX = '$';
+
+module.exports = create;
+module.exports.is = is;
+
+function create(parameters) {
+    var property = parameters.property !== undefined ? parameters.property : '$zoom';
+
+    var inner, outer;
+    if (!is(parameters)) {
+        inner = function() { return parameters; };
+        outer = function() { return inner; };
+        inner.isConstant = outer.isConstant = true;
+
+    } else if (property[0] === GLOBAL_ATTRIBUTE_PREFIX) {
+        outer = function(attributes) {
+            var value = evaluate(parameters, attributes[property]);
+            inner = function() { return value; };
+            inner.isConstant = true;
+            return inner;
+        };
+
```

Comment:
oh, I didn't realize isConstant applies to the partial function so I was thinking it would be `true`.

When creating the buffers we don't need to evaluate the scale at all if it's zoom based. Right now there is no way to tell unless you evaluate the first half. It's probably fine performance-wise to always evaluate it, but it might be nice to not need not.

I was trying https://github.com/mapbox/mapbox-gl-js/blob/83ca3f19cf8fd428f3fa1338b0209292eb9a97a3/js/data/line_bucket.js#L19 but I guess I need to evaluate the first half always